_example/cmd/database/others.go

```diff
@@ -46,13 +46,46 @@ func (db *Database) ListAvailableIdentityProviders() []resources.AvailableIdenti
 	return db.AvailableIdentityProviders
 }
 
-// ListUserResources returns the list of resources.
-func (db *Database) ListUserResources() []resources.Resource {
+// ListUserResources returns the list of resources. If entityType is nil, the
+// method returns all resources.
+func (db *Database) ListUserResources(entityType *string) []resources.Resource {
 	db.mutex.RLock()
 	defer db.mutex.RUnlock()
 	db.Load()
 
-	return db.Resources
+	if entityType == nil {
+		return db.Resources
```

Comment:
If `entityType` is nil, we only return the leaves of the graph? How come if entity type is specified do we return all elements in the graph that match?

Reply:
Good catch. Thanks.